client-app/shared/layout/components/search-bar/search-bar.vue

```diff
@@ -286,15 +286,15 @@ function reset() {
   hideSearchDropdown();
 }
 
-const searchProductsDebounced = useDebounceFn(() => {
+const searchProductsDebounced = useDebounceFn(async () => {
   if (!isApplied.value) {
-    void searchAndShowDropdownResults();
+    await searchAndShowDropdownResults();
   }
 }, SEARCH_BAR_DEBOUNCE_TIME);
 
-function onSearchPhraseChanged() {
+async function onSearchPhraseChanged() {
   hideSearchDropdown();
-  void searchProductsDebounced();
+  await searchProductsDebounced();
```

Comment:
return void everywhere and add rule for sonar to accept it.
It needs to fix the warning
![image](https://github.com/user-attachments/assets/bc3cceaf-48ad-4706-b271-c8a82203ca1b)

Reply:
reverted changes with "void"